src/core/thread/data_poll_manager.hpp

```diff
@@ -188,6 +188,12 @@ class DataPollManager : public InstanceLocator
      */
     void SendFastPolls(uint8_t aNumFastPolls);
 
+    /**
+     * This method asks data poll manager to stop fast poll mode after the expected response is received.
+     *
+     */
+    void StopFastPolls(void);
```

Comment:
The idea behind `SendFastPolls` was that different modules could request fast-polls at/around the same time. They each give a requested number of polls they want and the data poll manager code ensures to honor at least that many fast polls (it keeps track of current remaining fast poll count and update it when a new request comes in).

With `StopFastPolls()` we loose the capability to use it from multiple places in the code.

Reply:
This PR was submitted as there was only one type of message using `SendFastPolls` in existing code.
@abtink ,  Thanks for sharing the design purpose. How about adding one variable counting the number of callers that requesting `SendFastPolls`, and only successfully `StopFastPolls` when no callers? Or would you prefer simply remove such `StopFastPolls`?

Reply:
Separate this PR and move `StopFastPolls` to another one. We may continue the discuss there.

Reply:
Submitted https://github.com/openthread/openthread/pull/3763 to add `StopFastPolls()` support.

Reply:
Thanks. Good idea to have bug fix and enhancements in separate PRs.